frontend/device/android/device.lua

```diff
@@ -201,6 +204,9 @@ function Device:init()
                         end)
                     end
                 end
+            elseif ev.code == C.APP_CMD_STOP then
+                local Event = require("ui/event")
+                UIManager:broadcastEvent(Event:new("Suspend"))
```

Comment:
No weird side effects?

Reply:
Yeah, that one feels *super* funky. Ping @pazos

Reply:
No idea what the suspend event does. Also no idea how this fixes the issue @zwim has. I guess it is something tolino specific.

From the PoV of android apps onStop happens after onPause. Not exactly after that but a few minutes/hours or whatever. It makes sense for us to take onStop as a suspend state, because when it happens the execution of the program is stopped until the next onStart. No idea about side effects

Reply:
If the suspend event tries to write something to the blitbuffer it will fail because there's no window there. It shouldn't be problematic but can fill the log with https://github.com/koreader/koreader-base/blob/master/ffi/framebuffer_android.lua#L105

Reply:
I doubt the event does anything on Android; it's just a please double check and assure me type of remark. ;-)

Reply:
> No idea what the suspend event does. Also no idea how this fixes the issue @zwim has. I guess it is something tolino specific.


It is not Tolino specific. You can reproduce it on a mobile (e.g. Android11);
* Start KOReader,
* open a epub,
* enable both status bars, 
* enable autoupdate.

Wait until the clock changes.

Put the device to sleep and wait a bit more than 2 minutes (lets say 130sec).
Resume the device and you will see the old time in both status bars (for approx. 50s)

**What drove me almost crazy: The device MUST NOT be plugged in.**  

> 
> From the PoV of android apps onStop happens after onPause. Not exactly after that but a few minutes/hours or whatever. It makes sense for us to take onStop as a suspend state, because when it happens the execution of the program is stopped until the next onStart. No idea about side effects

Although I didn't notice any sideeffects the only thinkable (for me) could be that the scheduling of the clock refresh gets disrupted (which I did not notice).

Reply:
I'm pretty sure background tasks are frozen to half-death on modern Android versions, but, regardless, it appears to be mostly harmless, and mildly useful on funkier platforms.

As for @pazos's worries, at a quick glance, I don't think any Suspend handler do anything beyond "please stop doing this for a while" ^^.